src/main/java/com/minecolonies/coremod/commands/colonycommands/ShowColonyInfoCommand.java

```diff
@@ -56,11 +70,70 @@ public String getCommandUsage(@NotNull final ICommandSender sender)
         return super.getCommandUsage(sender) + "<ColonyId|OwnerName>";
     }
 
+    public void execute(@NotNull final MinecraftServer server, @NotNull final ICommandSender sender, @NotNull final ActionMenu actionMenu) throws CommandException
+    {
+        // See if we have a valid colony,
+        Colony colony = actionMenu.getColonyForArgument("colony");
+        EntityPlayer player = null;
+        if (null == colony)
+        {
+            // see if we have a valid player
+            player = actionMenu.getPlayerForArgument("player");
+            if (null != player)
+            {
+                final IColony iColony = ColonyManager.getIColonyByOwner(sender.getEntityWorld(), player);
+                if (iColony != null)
+                {
+                    if (!canPlayerUseCommand(player, SHOWCOLONYINFO, iColony.getID()))
+                    {
+                        sender.sendMessage(new TextComponentString(NOT_PERMITTED));
+                        return;
+                    }
+                    colony = ColonyManager.getColony(iColony.getID());
+                }
+            }
+        }
+
+        if (null == colony)
+        {
+            // see if we have a sender that is a valid player
+            if (sender instanceof EntityPlayer)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Merge this if statement with the enclosing one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1066)